tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2730,52 +2730,106 @@ class DoNotLogErrorsEndUpRaising(BaseChecker):
            Go through exception block and branches and ensure error hasn't been logged if exception is raised.
         """
-        # Return a list of exception blocks
-        except_block = node.handlers
-        # Iterate through each exception block
-        for nod in except_block:
-            # Get the nodes in each block (e.g. nodes Expr and Raise)
-            exception_block_body = nod.body
-            self.check_for_raise(exception_block_body)
-
-    def check_for_raise(self, node):
+        try:
+            # Return a list of exception blocks
+            except_block = node.handlers
+            # Iterate through each exception block
+            for nod in except_block:
+                # Get the nodes in each block (e.g. nodes Expr and Raise)
+                exception_block_body = nod.body
+                exception_name = nod.name.name
+                self.check_for_raise(exception_block_body, exception_name)
+        except:
+            pass
+
+    def check_for_raise(self, node, exception_name):
         """ Helper function - checks for instance of 'Raise' node
             Also checks 'If' and nested 'If' branches
         """
         for i in node:
             if isinstance(i, astroid.Raise):
-                self.check_for_logging(node)
+                self.check_for_logging(node, exception_name)
             # Check for any nested 'If' branches
             if isinstance(i, astroid.If):
-                self.check_for_raise(i.body)
+                self.check_for_raise(i.body, exception_name)
 
                 # Check any 'elif' or 'else' branches
-                self.check_for_raise(i.orelse)
+                self.check_for_raise(i.orelse, exception_name)
+
 
-    def check_for_logging(self, node):
-        """ Helper function - checks 'Expr' nodes to see if logging has occurred at 'warning' or 'error'
-            levels. Called from 'check_for_raise' function
+    def check_for_logging(self, node, exception_name):
+        """ Helper function - Called from 'check_for_raise' function
+            Checks if the same exception that's being raised is also being logged
         """
-        matches = [".warning", ".error", ".exception"]
-        for j in node:
-            if isinstance(j, astroid.Expr):
-                expression = j.as_string().lower()
-                if any(x in expression for x in matches):
-                    self.add_message(
-                        msgid=f"do-not-log-raised-errors",
-                        node=j,
-                        confidence=None,
-                    )
+
+        try:
+            # Find all function calls in the code
+            for j in node:
+                if isinstance(j, astroid.Expr) and isinstance(j.value.func, astroid.Attribute):
+                    # check that the attribute is a logging level
+                    if j.value.func.attrname in ["debug", "info", "warning", "error", "exception", "critical"]:
```

Comment:
there may be false positives but this is the best way to find logging level regardless of logging/logger/_logger name. we don't commonly have singular worded function calls imo